ta/arch/arm/link.mk

```diff
@@ -34,30 +34,45 @@ link-script-cppflags-$(sm) := -DASM=1 \
 -include $(link-script-dep$(sm))
 
 link-script-pp-makefiles$(sm) = $(filter-out %.d %.cmd,$(MAKEFILE_LIST))
+
+define gen-link-t
 $(link-script-pp$(sm)): $(link-script$(sm)) $(conf-file) $(link-script-pp-makefiles$(sm))
-	@$(cmd-echo-silent) '  CPP     $@'
-	$(q)mkdir -p $(dir $@)
-	$(q)$(CPP$(sm)) -Wp,-P,-MT,$@,-MD,$(link-script-dep$(sm)) \
-		$(link-script-cppflags-$(sm)) $< > $@
+	@$(cmd-echo-silent) '  CPP     $(link-script-pp$(sm))'
```

Comment:
Since you're evaluating this once before, replacing `$@` with `$$@` and the same for `$<` should minimize the changes.

Reply:
thanks, fixed

Reply:
BTW, interesting thing (I afraid I've already lost the track what is going on):
```
diff --git a/ta/arch/arm/link.mk b/ta/arch/arm/link.mk
index bd8765e..384984f 100644
--- a/ta/arch/arm/link.mk
+++ b/ta/arch/arm/link.mk
@@ -39,6 +39,8 @@ define gen-link-t
 $(link-script-pp$(sm)): $(link-script$(sm)) $(conf-file) $(link-script-pp-makefiles$(sm))
        @$(cmd-echo-silent) '  CPP     $$@'
        $(q)mkdir -p $(dir $$@)
+       @echo $(dir $(link-script-pp$(sm)))
+       @echo $(dir $$@)
        $(q)$(CPP$(sm)) -Wp,-P,-MT,$$@,-MD,$(link-script-dep$(sm)) \
                $(link-script-cppflags-$(sm)) $$< > $$@
 
```
Output:
```
out/arm/ta/avb/
./
```

Which basically means, that `$$@` != `$(link-script-pp$(sm))`.
How it can be explained? :(

Reply:
fixed